pwa-shell/src/ui/login/LoginFormContainer.js

```diff
@@ -4,11 +4,13 @@ import LoginForm from 'ui/login/LoginForm';
 
 const mapDispatchToProps = (dispatch, { history, location }) => ({
   onSubmit: data =>
-    dispatch(performLogin(data)).then(() => {
-      const redirectURI = new URLSearchParams(location.search).get(
-        'redirect_uri',
-      );
-      history.replace(redirectURI ? redirectURI : '/');
+    dispatch(performLogin(data)).then(res => {
+      if (res) {
```

Comment:
@jeffgo10 is it enough to have a response or we need a successful response (so, response.ok)?

Reply:
> @jeffgo10 is it enough to have a response or we need a successful response (so, response.ok)?

in login method in src/thunks, it only returns a response object once login is successful, otherwise it won't return anything. So technically it only needs an object response. though I thought about what you said, we can add response.ok condition to make it more secure.

Reply:
hi @gidesan, I have changed the return object from json to response object and added response.ok for validity. Pls review